flakeseries_table.go

```diff
@@ -0,0 +1,132 @@
+		panic("Can't set: not able to convert")
+	}
+	id, ok := m["Id"].(string)
+	if !ok {
+		panic("Id field is not present or is not a string")
+	}
+
+	timestamp, err := FlakeToTime(id)
+
+	if err != nil {
+		return nil, err
+	}
+
+	m[flakeTimestampFieldName] = timestamp
+	m[bucketFieldName] = o.bucket(timestamp.Unix())
+
+	return o.Table.Set(m), nil
+}
+
+func (o *flakeSeriesT) Update(id string, m map[string]interface{}) (Op, error) {
+	timestamp, err := FlakeToTime(id)
+	if err != nil {
+		return nil, err
+	}
+	bucket := o.bucket(timestamp.Unix())
+
+	return o.Where(Eq(bucketFieldName, bucket), Eq(flakeTimestampFieldName, timestamp), Eq("Id", id)).Update(m), nil
+}
+
+func (o *flakeSeriesT) Delete(id string) (Op, error) {
+	timestamp, err := FlakeToTime(id)
+	if err != nil {
+		return nil, err
+	}
+	bucket := o.bucket(timestamp.Unix())
+
+	return o.Where(Eq(bucketFieldName, bucket), Eq(flakeTimestampFieldName, timestamp), Eq("Id", id)).Delete(), nil
+}
+
+func (o *flakeSeriesT) Read(id string, pointer interface{}) (Op, error) {
+	timestamp, err := FlakeToTime(id)
+	if err != nil {
+		return nil, err
+	}
+	bucket := o.bucket(timestamp.Unix())
+	return o.Where(Eq(bucketFieldName, bucket), Eq(flakeTimestampFieldName, timestamp), Eq("Id", id)).ReadOne(pointer), nil
+}
+
+func (o *flakeSeriesT) List(startTime, endTime time.Time, pointerToASlice interface{}) Op {
+	buckets := o.buckets(startTime, endTime)
+	return o.Where(In(bucketFieldName, buckets...), GTE(flakeTimestampFieldName, startTime), LT(flakeTimestampFieldName, endTime)).Read(pointerToASlice)
+}
+
+// ListSince queries the flakeSeries for the items after the specified ID but within the time window,
+// if the time window is zero then it lists up until 5 minutes in the future
+func (o *flakeSeriesT) ListSince(id string, window time.Duration, pointerToASlice interface{}) (Op, error) {
+	startTime, err := FlakeToTime(id)
+	if err != nil {
+		return nil, err
+	}
+
+	var endTime time.Time
+
+	if window == 0 {
+		// no window set - so go up until 5 mins in the future
+		endTime = time.Now().Add(5 * time.Minute)
+	} else {
+		endTime = startTime.Add(window)
+	}
+	buckets := o.buckets(startTime, endTime)
+	return o.Where(In(bucketFieldName, buckets), GTE(flakeTimestampFieldName, startTime), LT(flakeTimestampFieldName, endTime), GT("Id", id)).Read(pointerToASlice), nil
+}
+
+func (o *flakeSeriesT) WithOptions(opt Options) FlakeSeriesTable {
+	return &flakeSeriesT{
+		Table:      o.Table.WithOptions(opt),
+		bucketSize: o.bucketSize,
+	}
+}
+
+func (o *flakeSeriesT) buckets(startTime, endTime time.Time) []interface{} {
+	buckets := []interface{}{}
+	start := o.bucket(startTime.Unix())
+	for i := start; i < endTime.Unix()*1000; i += int64(o.bucketSize/time.Second) * 1000 {
+		buckets = append(buckets, i)
+	}
+
+	return buckets
+}
+
+func (o *flakeSeriesT) bucket(secs int64) int64 {
+	return (secs - secs%int64(o.bucketSize/time.Second)) * 1000
+}
+
+func FlakeToTime(id string) (time.Time, error) {
```

Comment:
Yes, either that or make a `flake` library whose sole job is to deal with flake IDs (which both this and `idgen` can import).